internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -126,6 +140,112 @@ func (r *ClusterDeploymentsReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 	return r.updateState(ctx, clusterDeployment, cluster, nil)
 }
 
+func (r *ClusterDeploymentsReconciler) updateClusterMetadata(ctx context.Context, cluster *hivev1.ClusterDeployment, c *common.Cluster) error {
+
+	s, err := r.ensureAdminPasswordSecret(ctx, cluster, c)
+	if err != nil {
+		return err
+	}
+	k, err := r.ensureKubeConfigSecret(ctx, cluster, c)
+	if err != nil {
+		return err
+	}
+	cluster.Spec.Installed = true
+	cluster.Spec.ClusterMetadata = &hivev1.ClusterMetadata{
+		ClusterID: c.OpenshiftClusterID.String(),
+		InfraID:   c.ID.String(),
+		AdminPasswordSecretRef: corev1.LocalObjectReference{
+			Name: s.Name,
+		},
+		AdminKubeconfigSecretRef: corev1.LocalObjectReference{
+			Name: k.Name,
+		},
+	}
+	return r.Update(ctx, cluster)
+}
+
+func (r *ClusterDeploymentsReconciler) ensureAdminPasswordSecret(ctx context.Context, cluster *hivev1.ClusterDeployment, c *common.Cluster) (*corev1.Secret, error) {
+	s := &corev1.Secret{}
+	name := fmt.Sprintf(adminPasswordSecretStringTemplate, cluster.Name)
+	getErr := r.Get(ctx, types.NamespacedName{Namespace: cluster.Namespace, Name: name}, s)
+	if getErr == nil {
+		return s, nil
+	}
+	if k8serrors.IsNotFound(getErr) {
```

Comment:
same as for ensureKubeConfigSecret styling

Reply:
Done